BaseClasses.py

```diff
@@ -896,13 +897,13 @@ def connect(self, connecting_region: Region, name: Optional[str] = None,
             _exit.access_rule = rule
         _exit.connect(connecting_region)
     
-    def create_exit(self, name: str) -> Entrance:
+    def create_exit(self, name: str) -> entrance_type:
```

Comment:
This is not valid. Until we create a typing.Generic (or correctly typevar the `__class_getitem__`), we can not have a dynamic type here. Any class that inherits from Entrance - that is any Type valid for entrance_type - is also an Entrance, so returning Entrance here was correct.